### PR TITLE
[MGDAPI-4169] set default credential manager to minter and introduce metric

### PR DESCRIPTION
**Pull request by cecobask**
## Overview
Jira: https://issues.redhat.com/browse/MGDAPI-4169

## Verification
- Clone this branch
- Run `make cluster/prepare`
- Run `make run`
- Run `curl -s http://localhost:8383/metrics | grep "cro_sts_credentials_secret"` and see metric emitted:
```
cro_sts_credentials_secret{error="secrets \"sts-credentials\" not found",namespace="cloud-resource-operator"} 1
```

### CODE REVIEW

**Comment by bryankeane0** [comment]
/lgtm
Code looks good and metric was retrieved successfully after running cro on a cluster

File: pkg/providers/aws/credentials.go [hunk 1]
```diff
@@ -144,9 +145,11 @@ type CredentialManager interface {
 func NewCredentialManager(client client.Client) CredentialManager {
 	ns, _ := k8sutil.GetOperatorNamespace()
```
Comment (laurafitzgerald) [inline pkg/providers/aws/credentials.go:146]: We have an unhandled error here.

File: pkg/providers/aws/credentials_test.go [hunk 2]
```diff
@@ -73,8 +80,11 @@ func TestCredentialMinterManager_ReconcileProviderCredentials(t *testing.T) {
 	}
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
-			cm := NewCredentialManager(tc.client).(*CredentialMinterCredentialManager)
-			awsCreds, err := cm.ReconcileProviderCredentials(context.TODO(), "testNamespace")
+			cm, err := NewCredentialManager(tc.client)
+			if err != nil {
```
Comment (laurafitzgerald) [inline pkg/providers/aws/credentials_test.go:84]: The results in fatal here but not in a previous test. Any reason for the difference?

File: pkg/providers/aws/credentials.go [hunk 3]
```diff
@@ -141,13 +142,24 @@ type CredentialManager interface {
...
-		return NewCredentialMinterCredentialManager(client)
+func NewCredentialManager(client client.Client) (CredentialManager, error) {
+	ns, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		return nil, err
+	}
+	secret, err := getSTSCredentialsSecret(context.TODO(), client, ns)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			resources.SetSTSCredentialsSecretMetric(ns, err)
```
Comment (laurafitzgerald) [inline pkg/providers/aws/credentials.go:153]: I think this metric set could be moved outside of the isNotFound error as we are accepting that one as not an error effectively.

**Review by laurafitzgerald — CHANGES_REQUESTED** [review]
Left a couple comments inline but I the update to the provider is looking good.

File: pkg/providers/aws/credentials.go (hunk 3 above)
Comment (cecobask) [reply pkg/providers/aws/credentials.go]: Done 👍🏻

**Pull request merged**